### PR TITLE
[3006.x] Migrate service integration tests

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
See title.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by s0undt3ch — COMMENTED** [review]
Do these tests really need a master and minion running?

Could they potentially be turned into functional tests?

**Comment by MKLeb** [comment]
> Do these tests really need a master and minion running?
> 
> Could they potentially be turned into functional tests?

I have migrated them to functional tests.  I was thinking the same thing when doing the original migration.

**Review by twangboy — DISMISSED** [review]

File: tests/pytests/functional/modules/test_service.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        if pre_srv_status:
+            modules.service.enable(service_name)
+        else:
+            modules.service.disable(service_name)
+
+    if post_srv_enabled != pre_srv_enabled:
+        if pre_srv_enabled:
+            modules.service.enable(service_name)
+        else:
+            modules.service.disable(service_name)
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_service.py:67]:
```suggestion
def setup_service(service_name, modules):
    pre_srv_status = modules.service.status(service_name)
    pre_srv_enabled = service_name in modules.service.get_enabled()

    try:
        yield pre_srv_status
    finally:
        post_srv_status = modules.service.status(service_name)
        post_srv_enabled = service_name in modules.service.get_enabled()

        if post_srv_status != pre_srv_status:
            if pre_srv_status:
                modules.service.enable(service_name)
            else:
                modules.service.disable(service_name)

        if post_srv_enabled != pre_srv_enabled:
            if pre_srv_enabled:
                modules.service.enable(service_name)
            else:
                modules.service.disable(service_name)
```
We always want the `finally` block to run, even if tests fail.

File: tests/pytests/functional/modules/test_service.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+
+@pytest.mark.flaky(max_runs=4)
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_service.py:70]:
```suggestion
@pytest.mark.flaky(max_runs=4)
```
Are these tests still flaky?

File: tests/pytests/functional/states/test_service.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+@pytest.fixture(autouse=True)
+def setup_service(service_name, modules):
+    pre_srv_enabled = True if service_name in modules.service.get_enabled() else False
+    post_srv_disable = False
+    if not pre_srv_enabled:
+        modules.service.enable(service_name)
+        post_srv_disable = True
+    yield post_srv_disable
+    if post_srv_disable:
+        modules.service.disable(service_name)
```
Comment (s0undt3ch) [inline tests/pytests/functional/states/test_service.py:63]:
```suggestion
@pytest.fixture(autouse=True)
def setup_service(service_name, modules):
    pre_srv_enabled = True if service_name in modules.service.get_enabled() else False
    post_srv_disable = False
    if not pre_srv_enabled:
        modules.service.enable(service_name)
        post_srv_disable = True
    try:
        yield post_srv_disable
    finally:
        if post_srv_disable:
            modules.service.disable(service_name)
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**